hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/Job.groovy

```diff
@@ -330,6 +333,23 @@ class Job {
   }
 
   /**
+   * DSL method to specify required parameters for a job.
+   * Certain parameters must be set explicitly in oder for the job to run correctly in production.
+   * Specifying those parameters here provides a clear and explicit expectation for the users of the job.
+   * For example, specifying "required parameters" in a job template,
+   * and set those parameters in a job cloned from this template.
+   * This helps avoid production errors in an early stage, such as those caused by default parameters.
+   *
+   * @param args Args whose key 'parameters' has a list of value specifying the required parameters
+   */
+  @HadoopDslMethod
+  void required(Map args) {
+    LinkedHashSet<String> parameters = args.parameters;
```

Comment:
Yes, this will be a list. I was thinking about initializing this as a `LinkedHashSet` using the List the user passed so that it will take care of duplication at initialization. But maybe a List here would be more readable. I'll change that.